From 458726089fc0881f64f64e8a1e4f8f6ede6917d8 Mon Sep 17 00:00:00 2001
From: Ravi Vembu <raviv@motorola.com>
Date: Wed, 6 May 2015 16:28:39 -0500
Subject: [PATCH 254/959] IKSWL-13785: power: qpnp_smbcharger_mmi: Disable
 Charging until Heartbeat runs.

commit "31677ca21bc4a67683f60eb3a243ad49dcdb09a2" introduced an issue where
the stop/start Charging is done only when switching from COOL to NORMAL. This
causes a regression (charging doesn't work) in other paths where charger is
connected at powerup. Hence stop charging by default in the charger probe until
heartbeat runs.

Change-Id: I8d330ff2a9218adc1be5e4f44abb4a7a266df5df
Signed-off-by: Ravi Vembu <raviv@motorola.com>
Reviewed-on: http://gerrit.mot.com/745286
SME-Granted: SME Approvals Granted
SLTApproved: Slta Waiver <sltawvr@motorola.com>
Tested-by: Jira Key <jirakey@motorola.com>
Reviewed-by: Kenneth Kessler <kennykessler@motorola.com>
Reviewed-by: Vidyadhar Gurram <rnh786@motorola.com>
Submit-Approved: Jira Key <jirakey@motorola.com>
---
 drivers/power/qpnp-smbcharger-mmi.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/power/qpnp-smbcharger-mmi.c b/drivers/power/qpnp-smbcharger-mmi.c
index 55fdf7bb4d9..8a770396d8a 100644
--- a/drivers/power/qpnp-smbcharger-mmi.c
+++ b/drivers/power/qpnp-smbcharger-mmi.c
@@ -4770,6 +4770,7 @@ static int determine_initial_status(struct smbchg_chip *chip)
 	chg_term_handler(0, chip);
 	usbid_change_handler(0, chip);
 	src_detect_handler(0, chip);
+	smbchg_charging_en(chip, 0);
 
 	mutex_lock(&chip->usb_set_present_lock);
 	chip->usb_present = is_usb_present(chip);
-- 
2.11.0

